packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -0,0 +1,55 @@
+import regexParser from 'regex-parser';
+import { NextResponse, NextRequest } from 'next/server';
+import {
+  RedirectInfo,
+  GraphQLRedirectsService,
+  GraphQLRedirectsServiceConfig,
+  REDIRECT_TYPE_301,
+  REDIRECT_TYPE_302,
+  REDIRECT_TYPE_SERVER_TRANSFER,
+} from '@sitecore-jss/sitecore-jss/site';
+
+export class RedirectsMiddleware {
+  private redirectsService: GraphQLRedirectsService;
+
+  constructor(config: GraphQLRedirectsServiceConfig) {
+    this.redirectsService = new GraphQLRedirectsService(config);
+  }
```

Comment:
Add jsdoc comment for constructor.

Instead of making the fetch override occur in the sample app, I think we could do it here in the service? This middleware is specific to next.js, so it should be fine.

Also, I think we should create a dedicated type for config. It can be based on `GraphQLRedirectsServiceConfig`, minus the `fetch`.

Something like:
```suggestion
  export type RedirectsMiddlewareConfig = Omit<GraphQLRedirectsServiceConfig, 'fetch'>;

  constructor(config: RedirectsMiddlewareConfig) {
    // NOTE: we provide native fetch for compatibility on Next.js Edge Runtime (underlying default 'cross-fetch' is not currently compatible: https://github.com/lquixada/cross-fetch/issues/78)
    this.redirectsService = new GraphQLRedirectsService({ ...config, { fetch: fetch } });
  }
```